### ISSUE TITLE
BUG: Horizontal and Vertical arrow lines are not always displayed correctly in Obsidian

### ISSUE THREAD

**Issue by Epposan** [zsviczian/obsidian-excalidraw-plugin#1454]
**Environment**
SYSTEM INFO:
	Obsidian version: v1.4.16
	Installer version: v1.4.16
	Operating system: Windows 10 Pro 10.0.19045
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: none
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 11
	Plugins enabled: 11
		1: Homepage v3.6.0
		2: Find orphaned files and broken links v1.9.1
		3: Hover External Link Plugin v1.0.0
		4: Recent Files v1.3.8
		5: Filename Heading Sync v1.9.0
		6: Tag Wrangler v0.6.1
		7: Folder Note v0.7.3
		8: Advanced Tables v0.19.1
		9: Links v1.5.11
		10: Excalidraw v2.0.2
		11: Format Hotkeys v0.1.7

**Possible bug**
Depending on some of their properties, arrow lines that are drawn _really_ horizontal or vertical inside Excalidraw are not  visible in Obsidian when embedded _in line_

**To Reproduce**
1. As shown in the screenshot, draw some basic rectangles and connect them using arrows (without arrow head)
2. Set _Stroke width_, _Stroke style_, _Sloppiness_ and _Font size_ to their smallest/lowest value.
3. Add _Text labels_ to all arrows, except one

In the resulting _Obsidian view_, 
1. the line labelled '1' is not visible, only the text label.
2. the horizontal line below it, without text label, is visible 
3. the vertical line labelled '2' is not visible, only its label
4. the slightly slanted  line labelled '2', is visible
5. the left vertical line labelled '3' is not visible, only its label
6. the right vertical line labelled '3' is visible due to its increased _sloppiness_

**Expected behavior**
All lines should be equally visible.

**Screenshots**
_The Excalidraw view_
![excalidraw-view](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/23705690/b449c079-5d5c-4bb3-ac02-c9fa6bf65dfc)

_The Obsidian view_
![obsidian-view](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/23705690/9df2e668-6aa0-47d5-af96-72223129ce4c)

**Additional context**
n/a

**Comment by zsviczian** [comment]
is the an svg or png export or the screenshot from Obsidian.
Can you share the file that produced this?

**Comment by Epposan** [comment]
Thanks for getting back on the issue. The template bug report suggested screenshots, so that's what they are.
Please find attached the _test_ file that produced this, exported to _.svg_. (I could not attach the original _.md_)
![render-test-1 excalidraw](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/23705690/e253077f-e93e-450c-9071-e2ffdbd8a7c8)
HTH :-)

**Comment by eppmeister** [comment]
I know that you're extremely busy, but could you please spend a few minutes to see whether the issue reported is reproduceable/repeatable using the shared markdown file ?

**Comment by eppmeister** [comment]
Apparently not important enough ;-)
In any case: thank you for Excalidraw !

**Comment by zsviczian** [comment]
You are correct, apparently this is not that important. Simple reason is, I have drawn thousands of illustrations, and this never came up as an issue, thus it has never hurt enough to spend time on it.

If I open the sample file you attached, then indeed I am able to reproduce the issue. If I move the endpoint of the missing lines and save the drawing, the lines appear.
If I follow the steps in the first post, I am unable to reproduce the issue. Are you able to reproduce the issue following the steps above in 2.23.8? It could be that there was some issue in the past that resulted in corrupted arrow connections.

The issue can be reproduced on Excalidraw.com
I copied your sample image to Excalidraw.com and exported the image to SVG. The result is exactly the same.
https://excalidraw.com/#json=wjm2Fa1Ii7WzTntP55Ghy,XgzMffSVHPRQV-HicYjEHw

Thus this issue should be opened on Excalidraw.com, but only after you validate you have the exact steps to recreate the issue.

**Comment by zsviczian** [comment]
This is an even more minimalistic way to reproduce the issue. If I remove the text the line appears.

https://excalidraw.com/#json=ZmdMo36SyA2HLiMaKvvlN,IJxxm5341vb_ikFa4_IOaQ

**Comment by zsviczian** [comment]
Comparing outputs with and without text, the line is there in both, I guess the mask rectangle for the text hides somehow the line:

<img width="1866" height="681" alt="Image" src="https://github.com/user-attachments/assets/409459e6-4918-4556-b11a-7e6f6c7802c5" />

indeed, if I change this:
`<g mask="url(#mask-1DC3ncuEkudebfB-jGe6e)" stroke-linecap="round">`
to
`<g stroke-linecap="round">`
the line appears.

The issue is fascinating, but it is clearly an issue that needs to be fixed in the Excalidraw component, not the Obsidian plugin.

**Comment by zsviczian** [comment]
The bare minimum to reproduce the issue. A single arrow with a single bound character:

https://excalidraw.com/#json=59wzrDc9CmRukS53F4OPU,CXsIHVc6LXsWZyI537QkOA

**Comment by zsviczian** [comment]
opened: https://github.com/excalidraw/excalidraw/issues/11439

**Comment by zsviczian** [comment]
fixed in Obsidian Excalidraw 2.23.9

@eppmeister, thanks for raising the urgency of this. Unfortunately, with close to a 1000 open issues, older items get buried, especially if they are not something I experience in my own workflow.

Thanks to @krishhna24 for resolving it: https://github.com/excalidraw/excalidraw/pull/11441

**Comment by eppmeister** [comment]
All working fine now !

Thank you @zsviczian and @krishhna for fixing the issue !
